content/token-swap.md

```diff
@@ -0,0 +1,454 @@
+- Withdraw liquidity
+- Swap tokens
+
+# TL;DR
+
+- The **token swap program** is an SPL contract deployed to Mainnet and Devnet available for use by developers and protocols.
+- The program accepts six different **instructions**, all of which we will explore in this lesson.
+- Developers are able to create and use **liquidity pools** to swap between any SPL token that they wish.
+- The program uses a mathematical formula called "**curve**" to calculate the price of all trades. Curves aim to mimic normal market dynamics: for example, as traders buy a lot of one token type, the value of the other token type goes up.
+
+# Overview
+
+## Swap Pools
+
+Before we get into how to create and interact with swap pools on Solana, it’s important we understand the basics of what a swap pool even is. A swap pool is an aggregation of two different tokens, we’ll call them `TokenA` and `TokenB` for now, with the purpose of providing liquidity to facilitate exchanging A for B or B for A.
+
+Users provide liquidity to these pools by depositing their own tokens into each pool. These users are called liquidity providers. Once a liquidity provider (or LP) deposits some tokens to the swap pool, the LP-tokens are minted that represent their fractional ownership in the pool. LP’s are incentivized to provide liquidity because most swap pools charge a transaction fee to facilitate the swap. These transactions fees are then paid out to the LP’s in proportion to the amount of liquidity they are providing in the pool. When an LP is ready to withdraw their deposited liquidity, their LP-tokens are burned and tokens from the pool (proportional to the amount of LP-tokens burned) are sent to their wallet.
+
+The purpose of swap pools is to facilitate decentralized trade between users. The conventional way for users to execute trades like this is through a centralized exchange on a central limit [order book](https://www.investopedia.com/terms/o/order-book.asp). This is how you would buy a stock or future in traditional finance. Generally, this requires a trusted third-party intermediary, but due to the decentralized nature of cryptocurrency, we now have a new way to facilitate trade. [Project Serum](https://www.projectserum.com/) is an example of such a decentralized central limit order book built on Solana.
+
+Swap pools are completely decentralized. Anybody can issue instructions to the swap program to create a new swap pool between any SPL tokens they wish. Swap pools and AMMs (Automated Market Makers) are an extremely fascinating and complex topic of DeFi, it is outside the scope of this lesson to get into the nitty-gritty details of these topics here, but there is a ton of material out there available to you if you’re interested. The Solana Token Swap program was heavily inspired by [Uniswap](https://uniswap.org/) and [Balancer](https://balancer.fi/) for example, and more information is available in their excellent documentation.
+
+### Creating a Swap Pool
```

Comment:
This section is tough to digest. We ned to figure out a way to break it put differently I think.